package/MDAnalysis/analysis/encore/similarity.py

```diff
@@ -858,14 +848,22 @@ def hes(ensembles,
          [ 7032.19607004     0.        ]]
     """
 
+    if weights == 'mass':
+        weights = ['mass' for _ in range(len(ensembles))]
+    elif weights is not None:
+        if len(weights) != len(ensembles):
+            raise ValueError("need weights for every ensemble")
+    else:
+        weights = [None for _ in range(len(ensembles))]
```

Comment:
Yes I think so - since ensembles are independent MDAnalysis.Universe they are not technically required to have the same topology, so it would make sense to specify a weight/mass array for each of them. Thanks for these changes!